<?xml version="1.0"?>
<launch>

  <!-- Starts gazebo in paused mode -->
  <arg name="paused" default="true"/>

  <!-- Starts gazebo gui -->
  <arg name="gui" default="true"/>

  <arg name="robot_description" default="robot_description"/>
  <arg name="joint_limit_params" default="$(find ur_description)/config/ur10/joint_limits.yaml" doc="YAML file containing the joint limit values"/>
  <arg name="kinematics_params" default="$(find ur_description)/config/ur10/default_kinematics.yaml" doc="YAML file containing the robot's kinematic parameters. These will be different for each robot as they contain the robot's calibration."/>
  <arg name="physical_params" default="$(find ur_description)/config/ur10/physical_parameters.yaml" doc="YAML file containing the phycical parameters of the robots"/>
  <arg name="visual_params" default="$(find ur_description)/config/ur10/visual_parameters.yaml" doc="YAML file containing the visual model of the robots"/>
  <arg name="transmission_hw_interface" default="hardware_interface/PositionJointInterface" />
  <arg name="ft_sensed_joint" default="wrist_3_joint" />

  <!-- World file -->
  <arg name="world" default="ur10_world_with_ramp_and_box.world" />

  <!-- Startup simulated world -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find admittance_controller)/config/$(arg world)"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="gui" value="$(arg gui)"/>
  </include>

  <!-- Send robot urdf to param server -->
  <param name="robot_description" command="$(find xacro)/xacro '$(find force_torque_sensor)/urdf/ur10_with_ft_sensor.urdf.xacro' 
    joint_limit_params:=$(arg joint_limit_params)
    kinematics_params:=$(arg kinematics_params)
    physical_params:=$(arg physical_params)
    visual_params:=$(arg visual_params)
    transmission_hw_interface:=$(arg transmission_hw_interface) 
    ft_sensed_joint:=$(arg ft_sensed_joint)" />

  <param name="$(arg robot_description)_semantic" textfile="$(find ur10_moveit_config)/config/ur10.srdf" />

  <group ns="$(arg robot_description)_planning">
    <rosparam command="load" file="$(find ur10_moveit_config)/config/joint_limits.yaml"/>
  </group>

  <!-- Load default settings for kinematics; these settings are overridden by settings in a node's namespace -->
  <group ns="$(arg robot_description)_kinematics">
    <rosparam command="load" file="$(find ur10_moveit_config)/config/kinematics.yaml"/>
  </group>

  <!-- Push robot_description to factory and spawn robot in Gazebo -->
  <node name="spawn_gazebo_model" pkg="gazebo_ros" type="spawn_model" args="-urdf -param robot_description -model robot -z 1
    -J shoulder_pan_joint 1.3644
    -J shoulder_lift_joint -0.950874
    -J elbow_joint 1.944186
    -J wrist_1_joint 3.71903
    -J wrist_2_joint -1.570797
    -J wrist_3_joint -1.777184
    -unpause"
    respawn="false" output="screen"/>

</launch>